src/sentry/static/sentry/app/utils/withTeamProjects.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import {isEqual} from 'lodash';
+
+import {Team, Organization, Project} from 'app/types';
+import getDisplayName from 'app/utils/getDisplayName';
+import {Client} from 'app/api';
+
+// We require these props when using this HOC
+type DependentProps = {
+  api: Client;
+  organization: Organization;
+  teams: Team[];
+  loadingTeams: boolean;
+};
+
+type InjectedProjectsProps = {
+  projects: Project[];
+  loadingProjects: boolean;
+};
+
+type State = {
+  projects: Project[];
+  loadingProjects: boolean;
+};
+
+const withTeamProjects = <P extends InjectedProjectsProps>(
+  WrappedComponent: React.ComponentType<P>
+) =>
+  class extends React.Component<
+    Omit<P, keyof InjectedProjectsProps> &
+      Partial<InjectedProjectsProps> &
+      DependentProps,
+    State
+  > {
+    static displayName = `withTeamProjects(${getDisplayName(WrappedComponent)})`;
+
+    state = {
+      projects: [],
+      loadingProjects: true,
+    };
+
+    componentDidMount() {
+      if (!this.props.loadingTeams) {
+        this.fetchProjects();
+      }
+    }
+
+    componentDidUpdate(prevProps) {
+      if (!this.props.loadingTeams && !isEqual(prevProps.teams, this.props.teams)) {
+        this.fetchProjects();
+      }
+    }
+
+    async fetchProjects() {
+      this.setState({
+        loadingProjects: true,
+      });
+      const promises = this.getTeamsProjectPromises();
+      const projects: Project[][] = await Promise.all(promises);
```

Comment:
When these requests fail or are aborted is the error caught gracefully by the parent components?